R/RunModule.R

```diff
@@ -49,14 +49,33 @@ runModule <- function(analysisSpecifications, moduleIndex, executionSettings, ..
   saveRDS(jobContext, jobContextFileName)
 
 
+
   # Execute module using settings
   script <- "
     source('Main.R')
     jobContext <- readRDS(jobContextFileName)
-    connectionDetails <- keyring::key_get(jobContext$moduleExecutionSettings$connectionDetailsReference)
-    connectionDetails <- ParallelLogger::convertJsonToSettings(connectionDetails)
-    connectionDetails <- do.call(DatabaseConnector::createConnectionDetails, connectionDetails)
-    jobContext$moduleExecutionSettings$connectionDetails <- connectionDetails
+
+  "
+
+    # Set the connection information based on the type of execution being
+    # performed
+    if ("CdmExecutionSettings" %in% class(executionSettings)) {
+      script <- paste0(script,
+                       "connectionDetails <- keyring::key_get(jobContext$moduleExecutionSettings$connectionDetailsReference)
+                       connectionDetails <- ParallelLogger::convertJsonToSettings(connectionDetails)
+                       connectionDetails <- do.call(DatabaseConnector::createConnectionDetails, connectionDetails)
+                       jobContext$moduleExecutionSettings$connectionDetails <- connectionDetails")
+    } else {
```

Comment:
This assumes it is guaranteed to be either 'CdmExecutionSettings' or 'ResultsExecutionSettings'. Maybe we should explicitly test if it is 'ResultsExecutionSettings' here, and use a final else clause to throw an error?

Reply:
Agreed - I've used `checkmate::assert_mutli_class` at the top of the function call to enforce this and have also added a `stop` command for readability in that section of the code.